compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -152,7 +153,7 @@ class SyntheticMethods(thisPhase: DenotTransformer) {
      *  def equals(that: Any): Boolean =
      *    (this eq that) || {
      *      that match {
-     *        case x$0 @ (_: C) => this.x == this$0.x && this.y == x$0.y
+     *        case x$0 @ (_: C @unchecked) => this.x == this$0.x && this.y == x$0.y
```

Comment:
Add comment warning maintainers this code is fragile, since unchecked might hide type errors...